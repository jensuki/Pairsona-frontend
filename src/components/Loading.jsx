import '../styles/Loading.css';

const Loading = () => {
    return (
        <div className="loading-container" role="status">
            <i className="fas fa-spinner fa-spin fa-3x"></i>
        </div>
    );
};

export default Loading;