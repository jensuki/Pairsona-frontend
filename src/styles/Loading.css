.loading-container {
    position: fixed;
    top: 8rem;
    left: 0;
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 9999;
}

.fa-spinner {
    color: rgb(173, 200, 189);
}