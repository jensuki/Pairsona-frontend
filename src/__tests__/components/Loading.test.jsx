import { render } from "@testing-library/react";
import Loading from "../../components/Loading";
import { expect, test } from "vitest";

test("renders without crashing", () => {
    render(<Loading />);
})

test("matches snapshot", () => {
    const { asFragment } = render(<Loading />);
    expect(asFragment()).toMatchSnapshot();
})