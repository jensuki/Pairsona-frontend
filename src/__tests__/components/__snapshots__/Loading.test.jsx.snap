// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="loading-container"
    role="status"
  >
    <i
      class="fas fa-spinner fa-spin fa-3x"
    />
  </div>
</DocumentFragment>
`;
